AVLNode.h

```diff
@@ -1,20 +1,78 @@
+#include <iostream>
+using namespace std;
+
 #ifndef __AVL_NODE_H__
 #define __AVL_NODE_H__
 
+template<class T>
 class AVLNode
 {
 private:
-    int data;
+    T data;
     AVLNode * left;
     AVLNode * right;
 public:
-    AVLNode(int data, AVLNode*left, AVLNode*right);
+    enum Comparison { LESS, GREATER, EQUAL };
+    AVLNode(T data, AVLNode*left, AVLNode*right);
     ~AVLNode();
-    int getData() const;
+    T getData() const;
     AVLNode* getLeft() const;
     AVLNode* getRight() const;
-    void setData(int newData);
+    void setData(T newData);
     void setLeft(AVLNode * left);
     void setRight(AVLNode * right);
+    Comparison compare(T) const;
```

Comment:
Good implementation of compare... I think we will have to update this once we add the complementary-next-level features.